testing/cli/test_sct_run_batch.py

```diff
@@ -12,46 +13,43 @@
 from spinalcordtoolbox.scripts import sct_run_batch
 
 
-def write_dummy_script(script):
-    """
-    Dummy script that displays subject
-    :param script: NamedTemporaryFile object. Script to write
-    :return:
-    """
+@pytest.fixture
+def dummy_script(tmp_path):
+    """Dummy executable script that displays subject."""
+    path_out = str(tmp_path / "dummy_script.sh")
     script_text = """
     #!/bin/bash
     SUBJECT=$1
     echo $SUBJECT
     """
-    # indexing removes beginning newline
-    script.write(dedent(script_text)[1:])
-    script.flush()
+    with open(path_out, 'w') as script:
+        # indexing removes beginning newline
+        script.write(dedent(script_text)[1:])
+        script.flush()
+    return path_out
 
 
-def test_config_with_args_warning():
-    # This is formatted strangely because pep8 says so.
-    with \
-        NamedTemporaryFile('w', suffix='.json') as config,\
-            TemporaryDirectory() as data,\
-            TemporaryDirectory() as out:
+def test_config_with_args_warning(tmp_path):
+    data = str(tmp_path / 'data')
+    out = str(tmp_path / 'out')
+    config_path = str(tmp_path / 'config.json')
 
-        cnf_text = '{{"jobs": 1, "path_data": "{}", "path_output": "{}"}}'.format(data, out)
-        config.write(cnf_text)
-        config.flush()
+    with open(config_path, 'w') as config:
+        cnf_text = {"jobs": 1, "path_data": data, "path_output": out}
```

Comment:
This is a tiny nitpick, and I like that you're using `json.dump`, but maybe rename `cnf_text` since it's no longer `_text`.